packages/fxa-payments-server/server/lib/routes/legal-docs.js

```diff
@@ -0,0 +1,184 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const config = require('../../config');
+const joi = require('joi');
+const fetch = require('node-fetch');
+const Sentry = require('@sentry/node');
+const { cdnFqdn, resultsCacheLimit } = config.get('legalDocLinks');
+const acceptLanuageParser = require('accept-language-parser');
+const supportedLanguages = require('fxa-shared/l10n/supportedLanguages.json');
+const DEFAULT_LOCALE = 'en';
+const CACHE_DURATION = 24 * 60 * 60 * 1000; // a day in milliseconds
+const LANGUAGES_LIMIT = 3;
+
+/**
+ * A map for caching.  The key is a url and the value is an object with two properties:
+ *  - checkedOn: timestamp (in ms) when the last check against the url was performed
+ *  - result: boolean
+ *
+ * The size of this map will also be limited.  There are 71 supported languages
+ * and two docs per product (not price/plan).  With some generous rounding, we
+ * have 200 per product.  The current default is set to 65536, which should
+ * give us a decent runway.
+ */
+const cache = new Map();
+
+const getFullPath = (pathWithoutLocale, language) =>
+  // minimal handling of an errant '/'
+  pathWithoutLocale.endsWith('/')
+    ? `${pathWithoutLocale.substring(
+        0,
+        pathWithoutLocale.length - 1
+      )}.${language}.pdf`
+    : `${pathWithoutLocale}.${language}.pdf`;
+
+/**
+ * @param {obj} lang A language object from accept-language-parser
+ */
+const getLanguageTag = (lang) =>
+  lang.region ? `${lang.code}-${lang.region}` : lang.code;
+
+const isSupportedLanguage = (tag) => supportedLanguages.includes(tag);
+
+const isCacheExpired = (lastChecked) =>
+  Math.round(Date.now() - lastChecked) > CACHE_DURATION;
+
+/**
+ * Keeping this simple: if the cache's full, delete half of the entries.
+ * It mutates the map from the arguments directly.
+ */
+const checkAndShrinkCache = (cacheMap, limit) => {
+  if (limit > 0 && cacheMap.size >= limit) {
+    const half = Math.ceil(limit / 2);
+    const keys = Array.from(cacheMap.keys()).slice(0, half);
+    for (const k of keys) {
+      cacheMap.delete(k);
+    }
+  }
+};
+
+const isValidDocUrl = async (url) => {
+  // Check cache first
+  const cachedVal = cache.get(url);
+  if (cachedVal !== undefined) {
+    if (isCacheExpired(cachedVal['checkedOn'])) {
+      cache.delete(url);
+    } else {
+      return cachedVal['result'];
+    }
+  }
+
+  try {
+    const res = await fetch(url, { method: 'HEAD' });
```

Comment:
When I used the debugger, I found that the `url` passed into `isValidDocUrl` is: "http://localhost:5000/legal/mozilla_vpn_privacy_notice.en.pdf", and this `res` above's status code is 404, since indeed that URL is not valid (i.e. there is no `mozilla_vpn_privacy_notice.en.pdf` file in `./assets`).

Setup I used:
* `npx serve` from `./assets`
* update and restart payments with `PAYMENT_LEGAL_DOWNLOAD_FQDN=localhost`

I also tried "http://localhost:3031/legal-docs/?url=http://localhost:5000/legal/mozilla_vpn_privacy_notice/" with my browser's locale set to `"fr-FR"`, and got the same 404 error.